extensions/jms/src/test/java/org/mule/extensions/jms/test/queue/basic/JmsBaseQueuePublishAndConsumeTestCase.java

```diff
@@ -40,8 +41,22 @@ public void publishAndConsumeOnce() throws Exception {
   public void publishAndConsumeManyInOrder() throws Exception {
     destination = newDestination("publishAndConsumeManyInOrder");
     final List<String> messages = of("First", "Second", "Third", "Fourth");
-    messages.forEach(this::publish);
-    messages.forEach(payload -> assertThat(consume(), hasPayload(equalTo(payload))));
+
+    messages.forEach(message -> {
+      try {
+        publish(message);
+      } catch (Exception e) {
+        fail();
```

Comment:
this makes no sense. If it fails it should fail with the exception it threw not with an assertion which doesn't tell you what happened. Use a traditional `for` if the lambda gives you trouble with the checked exceptions